app/javascript/components/pages/PostIndex.js

```diff
@@ -1,65 +1,90 @@
 import React, { Component } from 'react'
 import { NavLink } from 'react-router-dom'
 import {
-	ListGroup,
-	ListGroupItemHeading,
-	ListGroupItemText,
-	Button,
+    ListGroup,
+    ListGroupItemHeading,
+    ListGroupItemText,
+    Button,
 } from 'reactstrap'
 import Footer from '../components/Footer'
 import galaxy from '../assets/galaxy.jpg'
-
-
 class PostIndex extends Component {
-	render() {
-		const { posts } = this.props
-		return (
-			<React.Fragment>
+    constructor(props){
+        super(props);
+        this.state = {}
+    }
+    onClick(post, newCount){
+        // updatepostcount pass in the post id and the newcount as args
+        // {123123123123: true, 3424: true}
+        const newState = {...this.state, [post.id]: true};
```

Comment:
this `3424: true` seems like strange modeling of data here. It's not clear what the boolean signifies. I see that in the render method it's being used to disable/enable the voting buttons. It might be a tiny bit more cumbersome to model the state this way, but perhaps doing something like `[post.id]: {isDisabled: true}` would provide more clarity on how the boolean is being used. It would also allow you had other state onClick using the object--making it more extensible.